plotly/presentation_objs/presentation_objs.py

```diff
@@ -0,0 +1,1199 @@
+"""
+dashboard_objs
+==========
+
+A module for creating and manipulating spectacle-presentation dashboards.
+"""
+
+import copy
+import random
+import re
+import string
+import warnings
+
+from plotly import exceptions, optional_imports
+
+IPython = optional_imports.get_module('IPython')
+
+HEIGHT = 700.0
+WIDTH = 1000.0
+
+CODEPANE_THEMES = ['tomorrow', 'tomorrowNight']
+VALID_STYLE_KEYS = ['fontFamily', 'fontSize', 'margin', 'position',
+                    'textAlign', 'opacity', 'color', 'fontStyle',
+                    'fontWeight', 'lineHeight', 'minWidth', 'textDecoration',
+                    'wordBreak']
+VALID_PROPS_KEYS = ['theme', 'listType', 'href']
+NEEDED_STYLE_KEYS = ['left', 'top', 'height', 'width']
+VALID_LANGUAGES = ['cpp', 'cs', 'css', 'fsharp', 'go', 'haskell', 'java',
+                   'javascript', 'jsx', 'julia', 'xml', 'matlab', 'php',
+                   'python', 'r', 'ruby', 'scala', 'sql', 'yaml']
+VALID_CLASS_STYLES = ['pictureleft', 'pictureright', 'picturemiddle',
+                      'pictureleft_tiled', 'pictureright_tiled']
+
+VALID_SLIDE_PROPS = ['class', 'transition', 'background-image',
+                     'background-position', 'background-repeat',
+                     'background-size', 'background_color']
+
+VALID_TRANSITIONS = ['slide',  'zoom', 'fade', 'spin']
+
+PRES_THEMES = ['moods', 'martik']
+
+VALID_GROUPTYPES = [
+    'leftgroup_v', 'rightgroup_v', 'middle', 'checkerboard_topleft',
+    'checkerboard_topright'
+]
+
+fontWeight_dict = {
+    'Thin': {'fontWeight': 100},
+    'Thin Italic': {'fontWeight': 100, 'fontStyle': 'italic'},
+    'Light': {'fontWeight': 300},
+    'Light Italic': {'fontWeight': 300, 'fontStyle': 'italic'},
+    'Regular': {'fontWeight': 400},
+    'Regular Italic': {'fontWeight': 400, 'fontStyle': 'italic'},
+    'Medium': {'fontWeight': 500},
+    'Medium Italic': {'fontWeight': 500, 'fontStyle': 'italic'},
+    'Bold': {'fontWeight': 700},
+    'Bold Italic': {'fontWeight': 700, 'fontStyle': 'italic'},
+    'Black': {'fontWeight': 900},
+    'Black Italic': {'fontWeight': 900, 'fontStyle': 'italic'},
+}
+
+NEEDED_STYLE_ERROR_MESSAGE = (
+    "'left', 'top', 'width', and 'height' parameters must be "
+    "set equal to a number (percentage) or a number with "
+    "'px' at the end of it. For example in "
+    "\n\n.left=10;top=50px{{TEXT}}\n\n the top left corner of "
```

Comment:
What's `{{TEXT}}` mean here?

Reply:
I just deleted that message. It's nested in a hidden function so we shouldn't have it at all

Reply:
Could you run flake8 on this file? That'll catch unused variables like this.

Reply:
`flake8` didn't catch unused variables but `pylint` did 👍